lib/dry/cli.rb

```diff
@@ -135,23 +132,50 @@ def parse(command, arguments, names)
 
       result = Parser.call(command, arguments, prog_name)
 
-      if result.help?
-        out.puts Banner.call(command, prog_name)
-        exit(0)
-      end
+      return help(command, prog_name) if result.help?
 
-      if result.error?
-        err.puts(result.error)
-        exit(1)
-      end
+      return error(result) if result.error?
+
+      [command_builder(command), result.arguments]
+    end
 
-      [command.new, result.arguments]
+    # Prepares command instance and injects context
+    #
+    # @param command [Dry::CLI::Command]
+    #
+    # @since 0.7.0
+    # @api private
+    def command_builder(command)
+      instance = command.is_a?(Class) ? command.new : command
+      instance
```

Comment:
```suggestion
      command.is_a?(Class) ? command.new : command
```

Reply:
this one is left like that because I will pass _context in the instance one PR later, so it will be
```ruby
instance = command.is_a?(Class) ? command.new : command
instance.set_instance_variable("@_context", Context.new...) 
instance
```

Reply:
Right. Let’s just save that change for when we do that, then? Ideally each PR/commit etc makes sense on its own. :)